### PR TITLE
feat: Hook into the CDA Parsing functions

### LINKED ISSUE #6715: feat: Add events to CDA Import
Adding events during the CDA import process would allow ill-formed CDA specs to be manipulated by a module instead of changing core functionality

### PR DESCRIPTION
**Pull request by robertdown**
Will resolve #6715. Add a new Event allowing hooks into `parseCDAEntryComponents()` for every component that gets parsed. The event dispatches the string of the component (encounter, allergy, medication, problems, etc), the OID, the component array, and the current state of `$templateData`.

`CdaTemplateParse->templateData` is always updated with the results of `$event->getTemplateData()` The event runs __after__ the core parsing function runs

### CODE REVIEW

File: src/Services/Cda/CdaTemplateParse.php [hunk 1]
```diff
@@ -22,11 +23,18 @@ class CdaTemplateParse
...
     protected $is_qrda_import;
 
+    /**
+     * @var \Symfony\Contracts\EventDispatcher\EventDispatcherInterface
+     */
+    private $ed;
+
     public function __construct()
     {
+        global $GLOBALS;
```
Comment (bradymiller) [inline src/Services/Cda/CdaTemplateParse.php:33]: this seems redundant since $GLOBALS is already a global

**Review by bradymiller — APPROVED** [review]

**Comment by bradymiller** [comment]
seems good to me :) looks like styling issues to fix. would have @sjpadgett give blessing before bringing in.

**Review by bradymiller — APPROVED** [review]

**Comment by bradymiller** [comment]
tenor99

**Comment by robertdown** [comment]
@sjpadgett you ok with these changes? It's testing well, I'm deep into manipulating a CDA doc that is a bit weird all using the hooks and it's working as expected.

**Comment by sjpadgett** [comment]
I haven't had time to look over but seems Brady and others are okay with so bring in and thanks. Kind of neat

**Pull request merged**